spec/registry/unicode.md

```diff
@@ -0,0 +1,63 @@
+# MessageFormat 2.0 Unicode Registry
+
+The `u:` namespace is reserved for use by the Unicode Consortium.
+
+## Options
+
+This section describes common options which each implementation SHOULD support
+for all _functions_ and _markup_.
+
+### `u:id`
+
+A string value that is included as an `id` or other suitable value
+in the formatted parts for the _placeholder_,
+or any other structured formatted results.
+
+Ignored when formatting a message to a string.
+
+Accepts string values, or values which can be stringified without error.
+For other values, a _Bad Option_ error is emitted
+and the `u:id` option is ignored.
+
+### `u:locale`
+
+A comma-delimited list of BCP 47 language tags,
+or an implementation-defined list of such tags.
+
+Replaces the _locale_ defined in the _function context_ for this _expression_.
+The value is ignored when set on _markup_.
```

Comment:
Why? Why not leave it up to the implementer?

Reply:
(Note: this comment is about line 28)

Reply:
Because that could introduce surprising differences in implementation behaviour and/or require us to define how markup open/close matching works. Consider this example, formatted as `en`:
```
French: {#span u:locale=fr}{$n :number}{/span}
English: {$n :number}
```

If we don't explicitly ignore `u:locale` and `u:dir` on markup, I would count it as likely that some implementations would format the first number in English, and others in French.

If that is desirable behaviour, then we would need to define open/close matching to ensure that the second number is still formatted in English.

Reply:
Wouldn't the `u:locale` only affect the markup placeholder? Why would it affect the enclosed `$n` placeholder?

Reply:
It's something that at least @mihnita has expressed some interest in, and if we leave it up to implementations, that becomes a possible interpretation of what happens when spanning markup has `u:` options set on it.

Reply:
In your example messages, the difference between the two is some markup. The formatting function should still format using the invoking (contextual) locale. For example, the invoking locale might be `fr-CA-u-nu-cans` and you wouldn't want the markup's `lang` attribute to remove the additional formatting (the example is contrived: I don't believe that the `Cans` script has its own digits). If you wanted to trim the locale, you should put it on the `:number` invocation. Otherwise markup can cause the formatting to change, which is spooky--and unlike anything else in MF2.

Reply:
This is now an error.